### PR TITLE
Quizzes API Support

### PR DESCRIPTION
**Pull request by mocca102**
wip

### CODE REVIEW

File: AutocompleteClient/FW/Logic/Request/Builder/CIOAnswerQuizQueryBuilder.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    
+    /**
+     Creata an Answer Quiz query builder
+     
+     - Parameters:
+        - quizId: The id of the quiz
+        - answers: A list of answers in the format ?a=[option_id_1]&a=[option_id_2] with one a parameter for    each question. The expected format for each parameter depends on the question type, as described below
+     */
+    
+    public init(quizId: String, answers: [String]) {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/Builder/CIOAnswerQuizQueryBuilder.swift:34]: Since `quizId` is the only required parameter, I think we can remove `answers` as a parameter from the `init` function.

File: AutocompleteClient/FW/Logic/Request/Builder/CIOAnswerQuizQueryBuilder.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ */
+public class CIOAnswerQuizQueryBuilder {
+
+    /**
+     The id of the quiz
+     */
+    let quizId: String
+
+    /**
+     A list of answers in the format ?a=[option_id_1]&a=[option_id_2] with one a parameter for each question. The expected format for each parameter depends on the question type, as described below
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/Builder/CIOAnswerQuizQueryBuilder.swift:22]: I think we should write a different description than using the one from the docs and link out to the documentation. Something like:

```
A list of answers that be in the following formats:
* Single Select: "answer1"
* Multiple Select: "answer1, answer2"
* Open Text: "true"
* Cover Page: "seen"

Please refer to <insert link to docs> for additional details
```

Lmk what you think

File: AutocompleteClient/FW/Logic/Request/Builder/CIOAnswerQuizQueryBuilder.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    var answers: [String]?
+    
+    /**
+     Creata an Answer Quiz query builder
+     
+     - Parameters:
+        - quizId: The id of the quiz
+        - answers: A list of answers in the format ?a=[option_id_1]&a=[option_id_2] with one a parameter for    each question. The expected format for each parameter depends on the question type, as described below
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/Builder/CIOAnswerQuizQueryBuilder.swift:31]: Should we shorten this to just be `answers: A list of answers`?

File: AutocompleteClient/FW/Logic/Request/CIOAutocompleteQuery.swift [hunk 4]
```diff
@@ -10,7 +10,7 @@ import Foundation
 
 /**
  Struct encapsulating the necessary and additional parameters required to execute an autocomplete query.
- */
+ */
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOAutocompleteQuery.swift:13]: Remove extra space

File: AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ Struct encapsulating the necessary and additional parameters required to execute a search query.
+ */
+public struct CIOQuizzesQuery: CIORequestData {
+    /**
+     The id of the quiz
+     */
+    public let quizId: String
+
+    /**
+     A list of answers in the format ?a=[option_id_1]&a=[option_id_2] with one a parameter for each question. The expected format for each parameter depends on the question type, as described below
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift:21]: Similar to the other file, I think we should write a different description than using the one from the docs and link out to the documentation. Something like:

```
A list of answers that be in the following formats:
* Single Select: "answer1"
* Multiple Select: "answer1, answer2"
* Open Text: "true"
* Cover Page: "seen"

Please refer to <insert link to docs> for additional details
```

Lmk what you think

File: AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    func url(with baseURL: String) -> String {
+        return String(format: Constants.Quiz.format, baseURL, quizId, finalize == true ? "finalize" : "next")
+    }
+
+    /**
+     Create a Answer Quiz request query object
+
+     - Parameters:
+        - quizId: The id of the quiz
+        - answers: A list of answers in the format ?a=[option_id_1]&a=[option_id_2] with one a parameter for each question. The expected format for    each parameter depends on the question type, as described below.
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift:36]: `answer: A list of answers`? 😄

File: AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+     ### Usage Example: ###
+     ```
+
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift:40]: Remove extra space

File: AutocompleteClient/FW/Logic/Result/CIONextQuestionData.swift [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            return nil
+        }
+
+        let options: [CIOQuestionOption]? = optionsObj?.compactMap { obj in return CIOQuestionOption(json: obj) }
+
+        self.id = id
+        self.title = title
+        self.description = description
+        self.ctaText = ctaText
+
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIONextQuestionData.swift:78]: Remove extra space

File: AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+ Struct encapsulating a question option attribute
+ */
+public struct CIOQuestionOptionAttribute {
+    public let name: String
+    public let value: String
+}
+
+
+/**
+ Struct encapsulating a question images
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift:21]: Should `images` be `option` or `option data`?

File: AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+ */
+public struct CIOQuestionOption {
+    public let id: String
+    public let value: String
+    public let attribute: CIOQuestionOptionAttribute?
+    public let images: CIOQuestionImages?
+}
+
+/**
+ Define a question images
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift:31]: Should images be option or option data?

File: AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+ */
+public extension CIOQuestionOption {
+    /**
+     Create a question images
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift:35]: Should images be option or option data?

File: AutocompleteClient/FW/Logic/Result/CIOQuizResultData.swift [hunk 12]
```diff
@@ -0,0 +1,40 @@
+//
+//  CIOQuizzesFinalizeData.swift
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuizResultData.swift:2]: Need to change this to be `CIOQuizResultData`

File: AutocompleteClient/FW/Logic/Result/Responses/CIOQuizzesResponse.swift [hunk 13]
```diff
@@ -0,0 +1,34 @@
+//
+//  CIOQuizResponse.swift
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/Responses/CIOQuizzesResponse.swift:2]: `CIOQuizzesResponse`

File: AutocompleteClientTests/FW/Logic/Request/AutocompleteQueryRequestBuilderTests.swift [hunk 14]
```diff
@@ -10,7 +10,7 @@ import XCTest
 @testable import ConstructorAutocomplete
 
 class AutocompleteQueryRequestBuilderTests: XCTestCase {
-
+
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/AutocompleteQueryRequestBuilderTests.swift:13]: Remove extra white space

File: AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+    
+    fileprivate let quizId: String = "1"
+    fileprivate let answers: [String] = ["1", "2,3", "seen", "true"]
+    fileprivate var endodedQuery: String = ""
+    fileprivate let testACKey: String = "abcdefgh123"
+    fileprivate var builder: RequestBuilder!
+
+    override func setUp() {
+        super.setUp()
+//        self.endodedQuery = answers.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed)!
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift:22]: Do we still need this?

File: AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+        self.builder = RequestBuilder(apiKey: self.testACKey, baseURL: "https://quizzes.cnstrc.com")
+    }
+
+    func testQuizzesQueryBuilder() {
+        let query = CIOQuizzesQuery(quizId: self.quizId, answers: self.answers)
+        builder.build(trackData: query)
+        let request = builder.getRequest()
+        let url = request.url!.absoluteString
+        print(url)
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift:31]: Remove print statement

File: AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+    fileprivate let testACKey: String = "abcdefgh123"
+    fileprivate var builder: RequestBuilder!
+
+    override func setUp() {
+        super.setUp()
+//        self.endodedQuery = answers.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed)!
+        self.builder = RequestBuilder(apiKey: self.testACKey, baseURL: "https://quizzes.cnstrc.com")
+    }
+
+    func testQuizzesQueryBuilder() {
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift:26]: Should this be called `testQuizzesQueryBuilder_next`?

File: AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+        let request = builder.getRequest()
+        let url = request.url!.absoluteString
+        print(url)
+        XCTAssertTrue(url.hasPrefix("https://quizzes.cnstrc.com/v1/quizzes/\(quizId)/next"))
+        XCTAssertTrue(url.contains("c=cioios-"), "URL should contain the version string.")
+        XCTAssertTrue(url.contains("key=\(testACKey)"), "URL should contain api key.")
+        XCTAssertEqual(request.httpMethod, "GET")
+    }
+    
+    func testQuizzesQueryBuilder_next() {
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift:38]: Should this be called `testQuizzesQueryBuilder_finalize`?

File: AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+        var query = CIOQuizzesQuery(quizId: self.quizId, answers: self.answers)
+        query.finalize = true
+        builder.build(trackData: query)
+        let request = builder.getRequest()
+        let url = request.url!.absoluteString
+        print(url)
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift:44]: Remove print statement

File: AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+    fileprivate var builder: RequestBuilder!
+
+    override func setUp() {
+        super.setUp()
+//        self.endodedQuery = answers.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed)!
+        self.builder = RequestBuilder(apiKey: self.testACKey, baseURL: "https://quizzes.cnstrc.com")
+    }
+
+    func testQuizzesQueryBuilder() {
+        let query = CIOQuizzesQuery(quizId: self.quizId, answers: self.answers)
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift:27]: This might need to change to use `setAnswers` after removing `answers` as a parameter from the init function

File: AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+        let url = request.url!.absoluteString
+        print(url)
+        XCTAssertTrue(url.hasPrefix("https://quizzes.cnstrc.com/v1/quizzes/\(quizId)/next"))
+        XCTAssertTrue(url.contains("c=cioios-"), "URL should contain the version string.")
+        XCTAssertTrue(url.contains("key=\(testACKey)"), "URL should contain api key.")
+        XCTAssertEqual(request.httpMethod, "GET")
+    }
+    
+    func testQuizzesQueryBuilder_next() {
+        var query = CIOQuizzesQuery(quizId: self.quizId, answers: self.answers)
```
Comment (jjl014) [inline AutocompleteClientTests/FW/Logic/Request/QuizzesQueryRequestBuilderTests.swift:39]: This might need to change to use setAnswers after removing answers as a parameter from the init function

File: AutocompleteClientTests/Resources/response_quizzes_next.swift [hunk 22]
```diff
@@ -0,0 +1 @@
+
```
Comment (jjl014) [inline AutocompleteClientTests/Resources/response_quizzes_next.swift:1]: Delete this file?

File: AutocompleteClient/Constants/Constants.swift [hunk 23]
```diff
@@ -215,4 +215,11 @@ struct Constants {
         static let id = "constructor.io/session/id"
         static let createdAt = "constructor.io/session/createdAt"
     }
+    
+    struct Quiz {
+        static let format = "%@/v1/quizzes/%@/%@"
+        static let quizId = "quiz_id"
+        static let answers = "a"
+        static let defaultAnswersValue = ""
```
Comment (jjl014) [inline AutocompleteClient/Constants/Constants.swift:223]: Doesn't seem like this is used anywhere, should we remove?

**Review by jjl014 — CHANGES_REQUESTED** [review]
Nice job! 💥  Overall this is looking pretty good. I left comments in various places. Let me know what you think.

I would also love to get @crgee1 and @stanlp1 review on this as well in case I missed something.

File: AutocompleteClient/FW/Logic/Request/Builder/CIOQuizzesQueryBuilder.swift [hunk 24]
```diff
@@ -0,0 +1,68 @@
...
+//  Constructor.io
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Request builder for creating an browse query.
```
Comment (stanlp1) [inline AutocompleteClient/FW/Logic/Request/Builder/CIOQuizzesQueryBuilder.swift:12]: Comment needs to be adjusted

File: AutocompleteClient/FW/Logic/Request/Builder/CIOQuizzesQueryBuilder.swift [hunk 25]
```diff
@@ -0,0 +1,68 @@
...
+     Please refer to "https://docs.constructor.io/rest_api/quiz/using_quizzes/" for additional details
+     */
+    var answers: [String]?
+    
+    /**
+     Create a Quizzes query builder
+     
+     - Parameters:
+        - quizId: The id of the quiz
+        - answers: A list of answers
```
Comment (stanlp1) [inline AutocompleteClient/FW/Logic/Request/Builder/CIOQuizzesQueryBuilder.swift:37]: Comment needs adjusting, answers is not part of init anymore.

File: AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift [hunk 26]
```diff
@@ -0,0 +1,57 @@
...
+//  Constructor.io
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Struct encapsulating the necessary and additional parameters required to execute a search query.
```
Comment (stanlp1) [inline AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift:12]: Update comment

File: AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift [hunk 27]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     A list of answers that be in the following formats:
+     * Single Select: "answer1"
+     * Multiple Select: "answer1, answer2"
+     * Open Text: "true"
+     * Cover Page: "seen"
+     
+     Please refer to "https://docs.constructor.io/rest_api/quiz/using_quizzes/" for additional details
+     */
+    public let answers: [String]?
```
Comment (stanlp1) [inline AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift:29]: Should we make this a var instead? 

Since let is constant. In case people want to reuse the same query and just change the answers.

File: AutocompleteClient/FW/Logic/Result/Responses/CIOQuizzesResponse.swift [hunk 28]
```diff
@@ -0,0 +1,34 @@
...
+//  AutocompleteClient
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Struct representing the search data response from the server.
```
Comment (stanlp1) [inline AutocompleteClient/FW/Logic/Result/Responses/CIOQuizzesResponse.swift:12]: Comment update.

**Review by stanlp1 — APPROVED** [review]
Tests pass lgtm, just minor comments on comments

File: AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift [hunk 29]
```diff
@@ -166,6 +168,28 @@ public class ConstructorIO: CIOSessionManagerDelegate {
...
+     ```
+     let quizzesQuery = CIOQuizzesQuery(quizId: "123", answers: ['a', 'b'])
+
+     constructorIO.quizzes(forQuery: quizzesQuery) { response in
+        let data = response.data!
+        let error = response.error!
+     }
+     ```
+     */
+    public func quizzes(forQuery query: CIOQuizzesQuery, completionHandler: @escaping  QuizzesQueryCompletionHandler) {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift:189]: Break this up into `getNextQuestion` and `getQuizResults`

File: AutocompleteClient/FW/Logic/Result/Responses/CIOQuizzesResponse.swift [hunk 30]
```diff
@@ -0,0 +1,34 @@
...
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Struct representing the quizzes data response from the server.
+ */
+public struct CIOQuizzesResponse {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/Responses/CIOQuizzesResponse.swift:14]: Break up into `CIONextQuestionResponse` and `CIOQuizResultsResponse`

File: AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+/**
+ Struct encapsulating the necessary and additional parameters required to execute a quizzes query.
+ */
+public struct CIOQuizzesQuery: CIORequestData {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOQuizzesQuery.swift:13]: Let's rename `CIOQuizzesQuery` to `CIOQuizQuery` to be consistent with the other SDKs

File: AutocompleteClient/FW/Logic/Result/CIONextQuestionData.swift [hunk 32]
```diff
@@ -0,0 +1,81 @@
...
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Struct encapsulating a result data object with additional information about the result
+ */
+public struct CIONextQuestionData {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIONextQuestionData.swift:14]: Let's rename this to `CIONextQuestion`

File: AutocompleteClient/FW/Logic/Result/CIONextQuestionData.swift [hunk 33]
```diff
@@ -0,0 +1,81 @@
...
+    public let options: [CIOQuestionOption]?
+    
+    /**
+     Input placeholder of the result question
+     */
+    public let inputPlaceholder: String?
+    
+}
+
+public extension CIONextQuestionData {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIONextQuestionData.swift:52]: => `CIONextQuestion`

File: AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift [hunk 34]
```diff
@@ -0,0 +1,51 @@
...
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Struct encapsulating a question option attribute
+ */
+public struct CIOQuestionOptionAttribute {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift:14]: Rename to `CIOQuizOptionAttribute`

File: AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift [hunk 35]
```diff
@@ -0,0 +1,51 @@
...
+    public let name: String
+    public let value: String
+}
+
+
+/**
+ Struct encapsulating a question option
+ */
+public struct CIOQuestionOption {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift:23]: Rename to `CIOQuizOption`

File: AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift [hunk 36]
```diff
@@ -0,0 +1,51 @@
...
+    public let id: String
+    public let value: String
+    public let attribute: CIOQuestionOptionAttribute?
+    public let images: CIOQuestionImages?
+}
+
+/**
+ Define a question option
+ */
+public extension CIOQuestionOption {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuestionOption.swift:33]: Rename to `CIOQuizOption`

File: AutocompleteClient/FW/Logic/Result/CIOQuizResultData.swift [hunk 37]
```diff
@@ -0,0 +1,40 @@
...
+//  http://constructor.io/
+//
+
+import Foundation
+
+
+/**
+ Struct encapsulating a result data object with additional information about the result
+ */
+public struct CIOQuizResultData {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuizResultData.swift:15]: Rename to `CIOQuizResults`

File: AutocompleteClient/FW/Logic/Result/CIOQuizResultData.swift [hunk 38]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    public let filterExpressions: [String: Any]
+
+    /**
+     Results Url
+     */
+    public let resultsUrl: String
+}
+
+public extension CIOQuizResultData {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Result/CIOQuizResultData.swift:27]: Rename to `CIOQuizResults`

File: AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift [hunk 39]
```diff
@@ -597,6 +621,29 @@ public class ConstructorIO: CIOSessionManagerDelegate {
             }
         }
     }
+    
+    private func executeQuizzes(_ request: URLRequest, completionHandler: @escaping QuizzesQueryCompletionHandler) {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Worker/ConstructorIO.swift:625]: We'll need to break this out into `executeGetNextQuestion` and `executeGetQuizResults` to follow the change above. I wonder if it's possible to just have this be `executeQuizRequest` and have it handle both endpoints? 🤔

File: AutocompleteClient/FW/Logic/Request/Builder/CIOQuizzesQueryBuilder.swift [hunk 40]
```diff
@@ -0,0 +1,67 @@
...
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Request builder for creating a quizzes query.
+ */
+public class CIOQuizzesQueryBuilder {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/Builder/CIOQuizzesQueryBuilder.swift:14]: => `CIOQuizQueryBuilder`

File: AutocompleteClient/FW/API/Parser/Quizzes/QuizzesResponseParser.swift [hunk 41]
```diff
@@ -0,0 +1,34 @@
...
+//  QuizzesResponseParser.swift
+//  AutocompleteClient
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+class QuizzesResponseParser: AbstractQuizzesResponseParser {
```
Comment (jjl014) [inline AutocompleteClient/FW/API/Parser/Quizzes/QuizzesResponseParser.swift:11]: => `QuizResponseParser`

File: AutocompleteClient/FW/API/Network/Response/QuizzesTaskResponse.swift [hunk 42]
```diff
@@ -0,0 +1,11 @@
...
+//  QuizzesTaskResponse.swift
+//  Constructor.io
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+public class QuizzesTaskResponse: TaskResponse<CIOQuizzesResponse, Error> {}
```
Comment (jjl014) [inline AutocompleteClient/FW/API/Network/Response/QuizzesTaskResponse.swift:11]: => `QuizTaskResponse`

File: AutocompleteClient/FW/API/Parser/Quizzes/AbstractQuizzesResponseParser.swift [hunk 43]
```diff
@@ -0,0 +1,14 @@
...
+//  AbstractQuizzesResponseParser.swift
+//  AutocompleteClient
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+protocol AbstractQuizzesResponseParser {
```
Comment (jjl014) [inline AutocompleteClient/FW/API/Parser/Quizzes/AbstractQuizzesResponseParser.swift:11]: => `AbstractQuizResponseParser`

We might need two different parsers for each endpoint now 🤔

File: AutocompleteClient/FW/DependencyInjection/DependencyContainer.swift [hunk 44]
```diff
@@ -33,6 +33,10 @@ class DependencyContainer {
     var recommendationsResponseParser: () -> AbstractRecommendationsResponseParser = {
         return RecommendationsResponseParser()
     }
+    
+    var quizzesResponseParser: () -> AbstractQuizzesResponseParser = {
```
Comment (jjl014) [inline AutocompleteClient/FW/DependencyInjection/DependencyContainer.swift:37]: => `quizResponseParser`

**Review by jjl014 — COMMENTED** [review]
Just added a bunch of comments on renaming things and what we need to split out into multiple functions. I also left a couple comments to change places where we have `quizzes` to just `quiz` to be consistent with the other SDKs. 👍

**Comment by mocca102** [comment]
Renaming is done sir @jjl014

File: AutocompleteClient/FW/Logic/Request/Builder/RequestBuilder.swift [hunk 45]
```diff
@@ -105,6 +107,26 @@ class RequestBuilder {
         return request
     }
 
+    final func getQuizRequest(finalize: Bool) -> URLRequest {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/Builder/RequestBuilder.swift:110]: Not sure if this is the best way to handle this, but this will make it so the consumer of the client doesn't have to pass in `finalize` as an argument when calling `getQuizResults`.

File: AutocompleteClient/FW/Logic/Request/CIOQuizQuery.swift [hunk 46]
```diff
@@ -0,0 +1,59 @@
...
+     
+     Please refer to "https://docs.constructor.io/rest_api/quiz/using_quizzes/" for additional details
+     */
+    public var answers: [String]?
+
+    func url(with baseURL: String) -> String {
+        return "" // Do nothing - Returns empty string to satsify protocol requirement
+    }
+
+    func urlWithFormat(baseURL: String, format: String) -> String {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOQuizQuery.swift:34]: Not sure if this is the best way to handle this, but this will make it so the consumer of the client doesn't have to pass in `finalize` as an argument when calling `getQuizResults`.

File: AutocompleteClient/FW/Logic/Request/CIOQuizQuery.swift [hunk 47]
```diff
@@ -0,0 +1,66 @@
...
+     Create a Quiz request query object
+
+     - Parameters:
+        - quizId: The id of the quiz
+        - answers: A list of answers
+        - versionId: The version of the quiz you would like to request
+
+     ### Usage Example: ###
+     ```
+     let quizQuery = CIOQuizQuery(quizId: "123", answers: [["1"], ["1","2"], versionId: "some-version-id")
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/CIOQuizQuery.swift:53]: Missing `]` at the end of answers param

**Review by stanlp1 — APPROVED** [review]
Looking good to me.

**Review by jjl014 — APPROVED** [review]
